Sources/TuistKit/Services/GraphImplicitImportLintService.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+import Path
+import Tools
+import TuistCore
+import TuistGenerator
+import TuistLoader
+import TuistSupport
+import XcodeGraph
+
+final class GraphImplicitImportLintService {
+    let importSourceCodeScanner: ImportSourceCodeScanner
+
+    init(importSourceCodeScanner: ImportSourceCodeScanner) {
+        self.importSourceCodeScanner = importSourceCodeScanner
+    }
+
+    func lint(graph: GraphTraverser, config _: Config) async throws -> [LintingIssue] {
+        let allTargets = graph
+            .allTargets()
+
+        let allTargetNames = Set(allTargets.map(\.target.productName))
+
+        var implicitTargetImports: [Target: Set<String>] = [:]
+        for project in graph.projects.values {
+            let allTargets = project.targets.values
+
+            for target in allTargets {
+                let targetImports = Set(try await imports(for: target))
+                let targetTuistDeclaredDependencies = target.dependencies.compactMap {
+                    switch $0 {
+                    case let .target(name: targetName, _):
+                        return project.targets[targetName]?.productName
+                    case let .project(target: targetName, path: projectPath, _):
+                        return graph.projects[projectPath]?.targets[targetName]?.productName
+                    default:
+                        return nil
+                    }
+                }
+                let implicitImports = targetImports.intersection(allTargetNames).subtracting(targetTuistDeclaredDependencies)
+                if !implicitImports.isEmpty {
+                    implicitTargetImports[target] = implicitImports
+                }
+            }
+        }
+        return implicitTargetImports.map { target, implicitDependencies in
+            return LintingIssue(
+                reason: "Target \(target.name) implicitly imports \(implicitDependencies.joined(separator: ", ")).",
+                severity: .warning
+            )
+        }
+    }
+
+    func imports(for target: XcodeGraph.Target) async throws -> Set<String> {
```

Comment:
This should be a private method and not directly tested at all.

Reply:
But it is much more easier to make tests on this method rather than create tests for lint function. What do you think maybe move it to other Service? Because I strongly believe that this part should be tested

Reply:
I don't understand why we need to test the `imports` method separately. What matters is the input of `lint` and the output of lint issues. This should be an implementation detail.

Reply:
I want to separate it, because it's a big part of the feature. And it's easier to test it separately. Now I made TargetImportScanner, which contains all the logic with scanning targets